server/chat-plugins/datasearch.ts

```diff
@@ -1725,20 +1727,22 @@ function runMovesearch(target: string, cmd: string, canAll: boolean, message: st
 				if (alts.gens[String(move.gen)]) continue;
 				if (Object.values(alts.gens).includes(false) && alts.gens[String(move.gen)] !== false) continue;
 			}
-			for (const recoveryType in alts.other) {
-				let hasRecovery = false;
-				if (recoveryType === "recovery") {
-					hasRecovery = !!move.drain || !!move.flags.heal;
-				} else if (recoveryType === "zrecovery") {
-					hasRecovery = (move.zMove?.effect === 'heal');
-				}
-				if (hasRecovery === alts.other[recoveryType]) {
-					matched = true;
-					break;
-				}
-			}
+			if (!zrecoveryUndefined || recoveryUndefined === false) {
```

Comment:
```suggestion
			if (!zrecoveryUndefined) {
```
If `something` is equal to `false` then `!something` will always be true.

Reply:
!recovery doesn't work with this change

![image](https://user-images.githubusercontent.com/72151514/105430059-6e61fb00-5c31-11eb-929b-695f255af593.png)

Reply:
Oh, I misread `recoveryUndefined` as `zrecoveryUndefined`.

Reply:
well, i tried !recoveryUndefined, and !pivot and !recoil still with errors. I tried recoveryUndefined (without !), and zrecovery doesn't works

Reply:
Is there a reason one of them is `=== false` and the other one is `!`? And why they're comparing with undefined instead of the usual existence check?

Reply:
Indeed. `=== false` is useless as long as both variables are `!`
However, i didn't get positive results on my local server without `undefined`